floss/language/go/extract.py

```diff
@@ -0,0 +1,431 @@
+                        continue
+
+        if section_name == ".rdata":
+            # Extract string blob in .rdata section
+            """
+            0048E620  5B 34 5D 75 69 6E 74 38  00 09 2A 5B 38 5D 69 6E  [4]uint8..*[8]in
+            0048E630  74 33 32 00 09 2A 5B 38  5D 75 69 6E 74 38 00 09  t32..*[8]uint8..
+            0048E640  2A 5B 5D 73 74 72 69 6E  67 00 09 2A 5B 5D 75 69  *[]string..*[]ui
+            0048E650  6E 74 31 36 00 09 2A 5B  5D 75 69 6E 74 33 32 00  nt16..*[]uint32.
+            0048E660  09 2A 5B 5D 75 69 6E 74  36 34 00 09 2A 63 68 61  .*[]uint64..*cha
+            0048E670  6E 20 69 6E 74 01 09 41  6E 6F 6E 79 6D 6F 75 73  n int..Anonymous
+            0048E680  01 09 43 61 6C 6C 53 6C  69 63 65 01 09 43 6C 65  ..CallSlice..Cle
+            0048E690  61 72 42 75 66 73 01 09  43 6F 6E 6E 65 63 74 45  arBufs..ConnectE
+            0048E6A0  78 01 09 46 74 72 75 6E  63 61 74 65 01 09 49 6E  x..Ftruncate..In
+            0048E6B0  74 65 72 66 61 63 65 01  09 4E 75 6D 4D 65 74 68  terface..NumMeth
+            0048E6C0  6F 64 01 09 50 72 65 63  69 73 69 6F 6E 01 09 52  od..Precision..R
+            """
+            section_va = section.VirtualAddress
+            section_size = section.SizeOfRawData
+            section_data = section.get_data(section_va, section_size)
+
+            blob_pattern = re.compile(b"(\x00|\x01)(.)", re.DOTALL)
+            for m in blob_pattern.finditer(section_data):
+                if m.group(2) != b"\x00":
+                    data = section_data[m.end() : m.end() + m.group(2)[0]]
+                    try:
+                        data = data.decode("utf-8")
+                        if str(data).isprintable() and data != "" and len(data) >= min_length:
+                            addr = m.start() + pe.OPTIONAL_HEADER.ImageBase + section_va
+                            yield StaticString(string=data, offset=addr, encoding=StringEncoding.ASCII)
+
+                    except UnicodeDecodeError:
+                        continue
+
+        if section_name == ".text":
+            # Extract string in .text section
+            section_va = section.VirtualAddress
+            section_size = section.SizeOfRawData
+            section_data = section.get_data(section_va, section_size)
+
+            for m in combinedregex.finditer(section_data):
+                for i in range(1, 8):
+                    try:
+                        tmp_string = m.group(i)
+                        if tmp_string != b"":
+                            try:
+                                decoded_string = tmp_string.decode("utf-8")
+                                if decoded_string.isprintable() and len(decoded_string) >= min_length:
+                                    addr = m.start() + pe.OPTIONAL_HEADER.ImageBase + section_va
+                                    yield StaticString(
+                                        string=decoded_string, offset=addr, encoding=StringEncoding.ASCII
```

Comment:
does this only yield the individual substrings?

Reply:
let's add a test case to prove that the full combined strings gets extracted

Reply:
"beauty of individual substrings" 🤣 
let's add an issue that this is a current shortcoming